river/ensemble/aggregated_mondrian_forest.py

```diff
@@ -0,0 +1,444 @@
+
+        # Checking the features consistency
+        self.check_features_consistency(x)
+
+        # Checking if the forest has been created
+        if self._forest is None:
+            self._initialize_trees()
+        # we fit all the trees using the new sample
+        for tree in self._forest:
+            tree.learn_one(x, y)
+        self.iteration += 1
+        return self
+
+    def predict_proba_one(self, x: dict) -> dict[int, float]:
+        """
+        Predicts the probability of each class for the sample x
+
+        Parameters
+        ----------
+        x: dict
+            Feature vector
+
+        Returns
+        -------
+        scores: dict
+        """
+
+        # Checking that the model has been trained once at least
+        if not self.is_trained():
+            raise Exception("No sample has been learnt yet. You need to train your model before making predictions.")
+
+        # initialize the scores
+        scores = {}
+        for j in range(self.n_classes):
+            scores[j] = 0
+
+        # Simply computes the prediction for each tree and average it
+        for tree in self._forest:
+            tree.use_aggregation = self.use_aggregation
+            predictions = tree.predict_proba_one(x)
+            for j in range(self.n_classes):
+                scores[j] += predictions[j] / self.n_estimators
+
+        return scores
+
+    @property
+    def n_classes(self):
+        """int: Number of expected classes in the labels."""
+        return self._n_classes
+
+    @n_classes.setter
+    def n_classes(self, val):
+        if self._forest:
+            raise ValueError(
+                "You cannot modify `n_classes` after calling `partial_fit`"
+            )
+        else:
+            if not isinstance(val, int):
+                raise ValueError("`n_classes` must be of type `int`")
+            elif val < 2:
+                raise ValueError("`n_classes` must be >= 2")
+            else:
+                self._n_classes = val
+
+    @property
+    def dirichlet(self):
+        """float or None: Regularization level of the class frequencies."""
+        return self._dirichlet
+
+    @dirichlet.setter
+    def dirichlet(self, val):
+        if self.is_trained():
+            raise ValueError(
+                "You cannot modify `dirichlet` after calling `learn_one`"
+            )
+        else:
+            if not isinstance(val, float):
+                raise ValueError("`dirichlet` must be of type `float`")
+            elif val <= 0:
+                raise ValueError("`dirichlet` must be > 0")
+            else:
+                self._dirichlet = val
+
+    @property
+    def loss(self):
+        """str: The loss used for the computation of the aggregation weights."""
+        return "log"
+
+    @loss.setter
+    def loss(self, val):
+        pass
+
+    def __repr__(self):
+        return f"AMFClassifier[n_classes={self.n_classes}; n_features={self.n_features}; n_models={self.n_estimators}]"
```

Comment:
You don't have to implement this. By default, a River model inherits from `base.Estimator`, which has a default `__repr__` method.

Reply:
Removed!